File: Tree/tree.py
class Node:
  def __init__(self, item):
    self.right = None
    self.left = None
    self.val = item

def inorderTraversal(root):
    if root:
        inorderTraversal(root.left)
        print(str(root.val) + " -> ", end="")
        inorderTraversal(root.right)

def preorderTraversal(root):
    if root:
        print(str(root.val) + " -> ", end="")
        preorderTraversal(root.left)
        preorderTraversal(root.right)

def postorderTraversal(root):
    if root:
        postorderTraversal(root.left)
        postorderTraversal(root.right)
        print(str(root.val) + " -> ", end="")

root = Node(1)
root.left = Node(12)
root.right = Node(9)

root.left.left = Node(5)
root.left.right = Node(6)

root.right.left = Node(23)
root.right.right = Node(42)

root.right.left.left = Node(56)
root.right.left.right = Node(99)

print("Inorder Traversal ")
inorderTraversal(root)
print("\nPreorder Traversal ")
preorderTraversal(root)
print("\nPostorder Traversal ")
postorderTraversal(root)

File: Tree/tree.c
#include <stdio.h>
#include <stdlib.h>

struct node
{
  int data;
  struct node *left;
  struct node *right;
};

struct node *nodeCreate(int value)
{
  struct node *nodeNew;
  nodeNew = (struct node *)malloc(sizeof(struct node));

  nodeNew->data = value;
  nodeNew->left = NULL;
  nodeNew->right = NULL;

  return nodeNew;
}

struct node* insertLeft(struct node* root, int value) {
  root->left = nodeCreate(value);
  return root->left;
}

struct node* insertRight(struct node* root, int value) {
  root->right = nodeCreate(value);
  return root->right;
}

void main()
{
  struct node *root;
  root = nodeCreate(4);
  /*
              4
            /   \
          NULL  NULL
    */
  root->left = nodeCreate(3);
  root->right = nodeCreate(7);
  /*
            4
         /     \
        3       7
      /   \    /  \
    NULL NULL NULL NULL
   */
}